### PR TITLE
Created ShopX-React Tool

### PR DESCRIPTION
**Pull request by v-sokolov**
Appending / Removing items in the main catalog based by design shopX web-site

### CODE REVIEW

File: ReactApps/reacttest/src/App.css [hunk 1]
```diff
@@ -0,0 +1,220 @@
+.App {
```
Comment (ghost) [inline ReactApps/reacttest/src/App.css:4]: Please split styles into different files for each component. Example : App.js - App.css, Wrapper.js - Wrapper.cs, Card.js- Card.css.

File: ReactApps/reacttest/src/card.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
+import React, {Component} from "react";
+
+class ItemCard extends Component {
```
Comment (ghost) [inline ReactApps/reacttest/src/card.js:3]: Do we really need state here ? It never changes.

File: ReactApps/reacttest/src/card.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+                        <button id="like"></button>
+                        <button id="compare"></button>
+                    </div>
+                    <div className="photo">
+                    </div>
+                </div>
+                <div className="sectionBody">
+                    <div className="text">
+                        <div>
+                            <p className="itemTitle">{this.props.title}</p>
```
Comment (ghost) [inline ReactApps/reacttest/src/card.js]: You can destructure props {title, oldPrice, newPrice} = this.props before return and use it.

File: ReactApps/reacttest/src/wrapper.jsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+import React, {Component} from "react";
+import ItemCard from "./card";
+
+class Wrapper extends Component {
+
+    constructor(props) {
+        super(props);
+        this.id = 0;
+        this.state = {
+            data: [
```
Comment (ghost) [inline ReactApps/reacttest/src/wrapper.jsx]: Do we need this initial state if its empty ? Please remove comments.

File: ReactApps/reacttest/src/wrapper.jsx [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+            // <ItemCard title="LavaVtase" oldPrice="68" newPrice ="35" id={this.id}/>,
+            // <ItemCard title="LavaVtase" oldPrice="68" newPrice ="35" id={this.id+1}/>,
+            // <ItemCard title="LavaVtase" oldPrice="68" newPrice ="35"/>
+            ]
+        };
+        //Бинд ивэнта клик к этому компоненту
```
Comment (ghost) [inline ReactApps/reacttest/src/wrapper.jsx]: Please use English for commenting.

File: ReactApps/reacttest/src/wrapper.jsx [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+                    <label htmlFor="title">Name of Product</label>
+                    <input type="text" onChange={this.handleChange} ref={(title) => this.name = title} id="title" placeholder="title"/>
+                    <label htmlFor="oldPrice">Old Price</label>
+                    <label htmlFor="newPrice">Current price with a discount</label>
+                    <input type="text" onChange={this.handleChange} ref={(oldPrice) => this.price = oldPrice} id="oldPrice" placeholder="oldPrice"/>
+                    <input type="text" onChange={this.handleChange} ref={(newPrice) => this.discount = newPrice} id="newPrice" placeholder="newPrice"/>
+                    <button id="add" onClick={this.handleClick}>Append</button>
+                </div>
+                <div className="catalog">
+                    {this.state.data}
```
Comment (ghost) [inline ReactApps/reacttest/src/wrapper.jsx]: Data should be raw data, not a component. So here you should create component for each data item in state.

**Review by ghost — CHANGES_REQUESTED** [review]

File: ReactApps/reacttest/src/wrapper.jsx (hunk 4 above)
Comment (v-sokolov) [reply ReactApps/reacttest/src/wrapper.jsx]: yes, it's need to render empty catalog

**Pull request closed without merging**